executers/utils/utils.go

```diff
@@ -55,22 +69,62 @@ func SetGoProxyWithApi(repoName string, details auth.ServiceDetails) error {
 		rtUrl.User = url.UserPassword(username, password)
 	}
 	rtUrl.Path += "api/go/" + repoName
-	err = os.Setenv(GOPROXY, rtUrl.String())
-	return errorutils.CheckError(err)
+	return rtUrl.String(), nil
+}
+
+// GetPackageVersion returns the matching version for the packageName string using the given artifactory details that provided.
```

Comment:
...given artifactory details that provided. --> ...using the Artifactory details that are provided.